### PR TITLE
12.0 afc 1950 Implementation of BOS V2 api

### PR DESCRIPTION
**Pull request by draal-af**

### CODE REVIEW

File: edi_af_ais_bos/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+{
+    'name': 'EDI AF AIS BOS',
+    'version': '12.0.2.0.0',
+    'category': 'edi',
+    'summary': """
+        Check if a postcode is valid for KROM.\n
+        Returns an enum for postnummer from IPF AIS BOS Regelverk V2 API.
+        """,
+    'licence': 'AGPL-3',
+    'description': """ """,
```
Comment (nils-vertel) [inline edi_af_ais_bos/__manifest__.py:31]: Add whats new in this version here

File: edi_af_ais_bos/data/edi_route_data.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        </record>
+
+        <record id="krom_postcode" model="edi.message.type">
+            <field name="name">AF asok: Get BOS KROM Postcode</field>
+            <field name="target">ais-bos-regelverk/v2/krom/kromtyp-for-postnummer</field>
+            <field name="type_mapping">{{url}}:{{port}}/{path}?client_id={{client}}&amp;client_secret={{secret}}&amp;postnummer={postnummer}</field>
+        </record>
+
+    </data>
+</openerp>
```
Comment (nils-vertel) [inline edi_af_ais_bos/data/edi_route_data.xml:18]: Missing newline at end of file

File: edi_af_ais_bos/data/edi_route_data.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<openerp>
```
Comment (nils-vertel) [inline edi_af_ais_bos/data/edi_route_data.xml:1]: Should be odoo?

File: edi_af_ais_bos/messages/af_get_bos.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from odoo import models, fields, api, _
```
Comment (nils-vertel) [inline edi_af_ais_bos/messages/af_get_bos.py:21]: Imports are wrongly ordered:

- Python standard lib
- Third party
- Own imports
Separate each group with a single new line, Sort each import within a group alphabetically

File: edi_af_ais_bos/messages/af_get_bos.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+_logger = logging.getLogger(__name__)
+
+
+class edi_message(models.Model):
+    _inherit = "edi.message"
+
+    @api.one
+    def unpack(self):
+        _logger.warning("KROM CHECK UNPACK")
```
Comment (nils-vertel) [inline edi_af_ais_bos/messages/af_get_bos.py:36]: Should this really be on Warning level, this should be info or debug IMHO

File: edi_af_ais_bos/messages/af_get_bos.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        if self.edi_type.id == self.env.ref('edi_af_ais_bos.krom_postcode').id:
+            # decode string and convert string to tuple, convert tuple to dict
+            body = json.loads(self.body.decode("utf-8"))
+
+            # Get values Krom
+            kromtype = body.get("kromTyp")
+            if kromtype:
+                self.env['res.partner'].browse(self.res_id).kromtype = kromtype
+            else:
```
Comment (nils-vertel) [inline edi_af_ais_bos/messages/af_get_bos.py:45]: Should we do something here?
As it is now the else clause can be removed

File: edi_af_ais_bos/messages/af_get_bos.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+                pass
+        else:
+            super(edi_message, self).unpack()
+
+    @api.one
+    def pack(self):
+        if self.edi_type.id == self.env.ref('edi_af_ais_bos.krom_postcode').id:
+            if not self.model_record or self.model_record._name != 'res.partner':
+                raise Warning("KROM: Attached record is not an res.partner'! {model}".format(
```
Comment (nils-vertel) [inline edi_af_ais_bos/messages/af_get_bos.py:54]: Warning is depricated use UserError instead

File: edi_af_ais_bos/messages/af_get_bos.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+                    model=self.model_record and self.model_record._name or None))
+
+            obj = self.model_record  # res.partner
+            given_address = obj.child_ids.filtered(lambda r: r.type == "given address")
+            if given_address:
+                postcode = given_address[0].zip
+            else:
+                raise Warning("Postcode field is empty or invalid")
```
Comment (nils-vertel) [inline edi_af_ais_bos/messages/af_get_bos.py:62]: See above

File: edi_af_ais_bos/models/ediroute.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from odoo import models, fields, api, _
```
Comment (nils-vertel) [inline edi_af_ais_bos/models/ediroute.py:21]: Wrongly sorted imports
Is all of these imports used?
Consider only keeping the standard imports (from odoo import models, fields, api, _
Write) (logging)

File: edi_af_ais_bos/views/edi_af_ais_bos_views.xml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  'route_type': 'edi_af_as_ais_bos',
+}
+
+msg = env['edi.message'].sudo().create(vals)
+msg.pack()
+route.run()
+  </field>
+ </record>
+
+</odoo>
```
Comment (nils-vertel) [inline edi_af_ais_bos/views/edi_af_ais_bos_views.xml:26]: No new line at end of file

File: edi_route_ipf/models/edi_route.py [hunk 11]
```diff
@@ -379,6 +393,7 @@ def get(self, message):
         # get list of occasions from res
 
         _logger.debug("ipf_rest.get() message.edi_type: %s" % message.edi_type)
+        _logger.debug("ipf_rest.get() message.edi_type: %s")
```
Comment (nils-vertel) [inline edi_route_ipf/models/edi_route.py:396]: Why the additional logger row here?

**Review by nils-vertel — CHANGES_REQUESTED** [review]

**Comment by daerio** [comment]
replaced with other PR

**Pull request closed without merging**